api/src/main/java/org/openmrs/api/db/hibernate/HibernateEncounterDAO.java

```diff
@@ -588,12 +590,26 @@ public Integer getEncountersByVisitsAndPatientCount(Patient patient, boolean inc
 		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Visit.class);
 		addEmptyVisitsByPatientCriteria(criteria, patient, includeVoided, query);
 		
+		//A hack to remove the order_by clauses as they cause error in PostgreSQL
+		Iterator<CriteriaImpl.OrderEntry> orderIter = ((CriteriaImpl) criteria).iterateOrderings();
```

Comment:
Why use a hack?

Reply:
There is no direct way to remove the order clause from criteria once it gets added( [docs](https://docs.jboss.org/hibernate/orm/3.3/reference/en/html/querycriteria.html) ).  I found this solution [here](https://stackoverflow.com/questions/2370221/remove-dynamically-an-ordering-to-the-result-set-in-org-hibernate-criteria).